src/main/kotlin/com/openlattice/mechanic/upgrades/AdjustNCRICDataDateTimeHashes.kt

```diff
@@ -0,0 +1,147 @@
+        val propertyTypesByFqn = toolbox.propertyTypes.values.associate { it.type.fullQualifiedNameAsString to it.id }
+
+        entitySetsToPropertyTypes.entries.stream().parallel().forEach {
+            val entitySetName = it.key
+
+            val entitySet = entitySetsByName.getValue(entitySetName)
+            val propertyTypeId = propertyTypesByFqn.getValue(it.value)
+
+            logger.info("About to update find ids needing updating from entity set $entitySetName")
+
+            toolbox.hds.connection.use { conn ->
+                conn.prepareStatement(createTempIdsTableSql(it.key)).use { ps ->
+                    ps.setObject(1, entitySet.id)
+                    ps.setObject(2, propertyTypeId)
+                    ps.setArray(3, PostgresArrays.createIntArray(conn, entitySet.partitions))
+
+                    ps.execute()
+                }
+            }
+
+            logger.info("Finished identifying ids needing updating from entity set $entitySetName. About to update values.")
+
+            try {
+                var insertCounter = 0
+                var insertCount = 1
+
+                val readBatchSql = getReadBatchSql(it.key)
+
+                while (insertCount > 0) {
+
+                    toolbox.hds.connection.use { conn ->
+                        conn.prepareStatement(UPDATE_SQL).use { ps ->
+
+                            BasePostgresIterable(StatementHolderSupplier(toolbox.hds, readBatchSql)) { rs ->
+
+                                val dateTimeObj = rs.getObject(DATETIME_COL)
+                                val odt = OffsetDateTime.ofInstant((dateTimeObj as Timestamp).toInstant(), ZoneOffset.UTC)
+
+                                ps.setObject(1, PostgresDataHasher.hashObject(odt, EdmPrimitiveTypeKind.DateTimeOffset))
+                                ps.setObject(2, entitySet.id)
+                                ps.setObject(3, ResultSetAdapters.id(rs))
+                                ps.setObject(4, propertyTypeId)
+                                ps.setInt(5, ResultSetAdapters.partition(rs))
+                                ps.setObject(6, dateTimeObj)
+
+                                ps.addBatch()
+
+                                1
+                            }.count()
+
+                            insertCount = ps.executeBatch().sum()
+                            logger.info("Updated $insertCount property hashes for entity set $entitySetName.")
+                            insertCounter += insertCount
+                        }
+
+                    }
+                }
+
+                logger.info("Finished migrating $insertCounter hashes for entity set $entitySetName")
+
+            } catch (e: Exception) {
+                logger.info("Something bad happened while updating entity set $entitySetName :(", e)
+            }
+
+        }
+
+        return true
+    }
+
+    fun createTempIdsTableSql(entitySetName: String): String {
```

Comment:
Minor thing, but this should probably be deleted at the end

Reply:
I figured it's easy enough to manually delete them and it might be useful to be able to inspect them afterwards as a sanity check

Reply:
👍